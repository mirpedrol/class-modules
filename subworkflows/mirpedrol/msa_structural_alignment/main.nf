include { FOLDMASON_EASYMSA } from '../../../modules/nf-core/foldmason/easymsa/main'
include { MTMALIGN_ALIGN } from '../../../modules/nf-core/mtmalign/align/main'


workflow MSA_STRUCTURAL_ALIGNMENT {

    take:
    ch_pdbs

    main:
    def ch_out_alignment = Channel.empty()
    def ch_out_versions = Channel.empty()

    ch_pdbs
        .branch {
            meta, pdbs, tool ->
                foldmason_easymsa: tool == "foldmason_easymsa"
                    return [ meta, pdbs ]
                mtmalign_align: tool == "mtmalign_align"
                    return [ meta, pdbs ]
        }
        .set { ch_pdbs_branch }

    FOLDMASON_EASYMSA( ch_pdbs_branch.foldmason_easymsa, [], [] )
    ch_out_alignment = ch_out_alignment.mix(FOLDMASON_EASYMSA.out.msa_aa)
    ch_out_versions = ch_out_versions.mix(FOLDMASON_EASYMSA.out.versions)

    MTMALIGN_ALIGN( ch_pdbs_branch.mtmalign_align, [] )
    ch_out_alignment = ch_out_alignment.mix(MTMALIGN_ALIGN.out.alignment)
    ch_out_versions = ch_out_versions.mix(MTMALIGN_ALIGN.out.versions)



    emit:
    alignment = ch_out_alignment
    versions = ch_out_versions

}

