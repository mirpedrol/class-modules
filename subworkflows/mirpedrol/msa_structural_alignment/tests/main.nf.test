nextflow_workflow {

    name "Test Subworkflow MSA_STRUCTURAL_ALIGNMENT"
    script "../main.nf"
    workflow "MSA_STRUCTURAL_ALIGNMENT"

    tag "subworkflows"
    tag "subworkflows_mirpedrol"
    tag "subworkflows/msa_structural_alignment"
    tag "subworkflows/../../modules/nf-core/foldmason/easymsa"
    tag "subworkflows/../../modules/nf-core/mtmalign/align"


    test("foldmason/easymsa") {

        when {
            workflow {
                """
                input[0] = Channel.of( [[ id:'test_tree' ], [file(params.modules_testdata_base_path + 'proteomics/pdb/1tim.pdb', checkIfExists: true), file(params.modules_testdata_base_path + 'proteomics/pdb/8tim.pdb', checkIfExists: true)], 'foldmason_easymsa'] )
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match("foldmason/easymsa") },
            )
        }
    }

    test("mtmalign/align") {

        when {
            workflow {
                """
                input[0] = Channel.of( [[ id:'test_tree' ], [file(params.modules_testdata_base_path + 'proteomics/pdb/1tim.pdb', checkIfExists: true), file(params.modules_testdata_base_path + 'proteomics/pdb/8tim.pdb', checkIfExists: true)], 'mtmalign_align'] )
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match("mtmalign/align") },
            )
        }
    }


}