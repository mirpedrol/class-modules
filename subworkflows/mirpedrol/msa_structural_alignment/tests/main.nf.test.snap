{
    "foldmason/easymsa": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_tree"
                        },
                        "test_tree_aa.fa:md5,2c20e002e5efd0d43cde8c5e896e998e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,079537a12ecb1a4e69170051f18a9c73"
                ],
                "alignment": [
                    [
                        {
                            "id": "test_tree"
                        },
                        "test_tree_aa.fa:md5,2c20e002e5efd0d43cde8c5e896e998e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,079537a12ecb1a4e69170051f18a9c73"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-01T12:22:19.477678328"
    },
    "mtmalign/align": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_tree"
                        },
                        "test_tree.aln:md5,393e2f3480c2daddceef798c03201a9d"
                    ]
                ],
                "1": [
                    "versions.yml:md5,3c92191a04b0f0a03555dbb7fed62a5c"
                ],
                "alignment": [
                    [
                        {
                            "id": "test_tree"
                        },
                        "test_tree.aln:md5,393e2f3480c2daddceef798c03201a9d"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3c92191a04b0f0a03555dbb7fed62a5c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-01T12:22:27.340342692"
    }
}
